askama_derive/src/lib.rs

```diff
@@ -2,9 +2,628 @@
 #![deny(elided_lifetimes_in_paths)]
 #![deny(unreachable_pub)]
 
+use std::borrow::Cow;
+use std::collections::{BTreeMap, HashSet};
+use std::convert::TryFrom;
+use std::path::{Path, PathBuf};
+use std::{env, fmt, fs};
+
 use proc_macro::TokenStream;
+use proc_macro2::Span;
+#[cfg(feature = "serde")]
+use serde::Deserialize;
+
+mod generator;
+mod heritage;
+mod input;
+mod parser;
 
+/// The actual implementation for askama_derive::Template
 #[proc_macro_derive(Template, attributes(template))]
 pub fn derive_template(input: TokenStream) -> TokenStream {
-    askama_shared::derive_template(input.into()).into()
+    generator::derive_template(input)
+}
+
+#[derive(Debug)]
```

Comment:
Would you mind doing a separate commit before this one that moves all the config stuff into a separate `config` module? I think that would make a lot of this stuff clearer.